pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "jquery",
+    "base1/cockpit",
+], function($, cockpit) {
+
+"use strict";
+
+var _ = cockpit.gettext;
+
+var client = { };
+
+client.subscriptionStatus = {
+    serviceStatus: undefined,
+    status: undefined,
+    products: [],
+    error: undefined,
+};
+
+// callback that is used to notify when data has changed
+var updateCallback;
+
+// DBUS service
+var service;
+
+var needRender = function() {
```

Comment:
This form of function makes it seem like the variable needRender will change. How about just the simpler ```function needRender()```